lib/src/common/enums.dart

```diff
@@ -0,0 +1,430 @@
+part of tiled;
+
+enum MapOrientation { orthogonal, isometric, staggered, hexagonal }
+
+extension MapOrientationExtension on MapOrientation {
+  String get name {
+    switch (this) {
+      case MapOrientation.orthogonal:
+        return 'orthogonal';
+      case MapOrientation.isometric:
+        return 'isometric';
+      case MapOrientation.staggered:
+        return 'staggered';
+      case MapOrientation.hexagonal:
+        return 'hexagonal';
+    }
+  }
+}
+
+extension MapOrientationParser on Parser {
+  MapOrientation? getMapOrientationOrNull(
+    String name, {
+    MapOrientation? defaults,
+  }) {
+    return getRawEnumOrNull(
+      MapOrientation.values,
+      (e) => e.name,
+      name,
+      defaults,
+    );
+  }
+
+  MapOrientation getMapOrientation(String name, {MapOrientation? defaults}) {
+    return getRawEnum(MapOrientation.values, (e) => e.name, name, defaults);
+  }
+}
+
+enum RenderOrder { rightDown, rightUp, leftDown, leftUp }
+
+extension RenderOrderExtension on RenderOrder {
+  String get name {
+    switch (this) {
+      case RenderOrder.rightDown:
+        return 'right-down';
+      case RenderOrder.rightUp:
+        return 'right-up';
+      case RenderOrder.leftDown:
+        return 'left-down';
+      case RenderOrder.leftUp:
+        return 'left-up';
+    }
+  }
+}
+
+extension RenderOrderParser on Parser {
+  RenderOrder? getRenderOrderOrNull(String name, {RenderOrder? defaults}) {
+    return getRawEnumOrNull(RenderOrder.values, (e) => e.name, name, defaults);
+  }
+
+  RenderOrder getRenderOrder(String name, {RenderOrder? defaults}) {
+    return getRawEnum(RenderOrder.values, (e) => e.name, name, defaults);
+  }
+}
+
+enum StaggerAxis { x, y }
+
+extension StaggerAxisExtension on StaggerAxis {
+  String get name {
+    switch (this) {
+      case StaggerAxis.x:
+        return 'x';
+      case StaggerAxis.y:
+        return 'y';
+    }
+  }
+}
+
+extension StaggerAxisParser on Parser {
+  StaggerAxis? getStaggerAxisOrNull(String name, {StaggerAxis? defaults}) {
+    return getRawEnumOrNull(StaggerAxis.values, (e) => e.name, name, defaults);
+  }
+
+  StaggerAxis getStaggerAxis(String name, {StaggerAxis? defaults}) {
+    return getRawEnum(StaggerAxis.values, (e) => e.name, name, defaults);
+  }
+}
+
+enum StaggerIndex { odd, even }
+
+extension StaggerIndexExtension on StaggerIndex {
+  String get name {
+    switch (this) {
+      case StaggerIndex.odd:
+        return 'odd';
+      case StaggerIndex.even:
+        return 'even';
+    }
+  }
+}
+
+extension StaggerIndexParser on Parser {
+  StaggerIndex? getStaggerIndexOrNull(String name, {StaggerIndex? defaults}) {
+    return getRawEnumOrNull(StaggerIndex.values, (e) => e.name, name, defaults);
+  }
+
+  StaggerIndex getStaggerIndex(String name, {StaggerIndex? defaults}) {
+    return getRawEnum(StaggerIndex.values, (e) => e.name, name, defaults);
+  }
+}
+
+enum VAlign { center, bottom, top }
```

Comment:
I would spell these out, even though they get quite long

Reply:
I think I prefer to be consistent with the terminology tiled uses, no? since it's pretty clear what it means

Reply:
Aah didn't know they had that terminology, then definitely use that.